src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1229,23 +1233,126 @@ private int SkipBlockComment(int i)
+                case 'r': return "\r";
+                case 't': return "\t";
+                case 'u': return ScanUnicodeEscapeSequence();
+                case 'v': return "\v";
+                default: return c.ToString();
+            }
+        }
+
+        private string ScanUnicodeEscapeSequence()
+        {
+            int escSeqStartIndex = _currentIndex - 2;
+            int maxNumberOfHexDigits = 6;
+            bool isTerminated = false;
+
+            char c = GetChar();
+            if (c != '{')
+            {
+                UngetChar();
+
+                IScriptExtent errorExtent = NewScriptExtent(escSeqStartIndex, _currentIndex);
+                ReportError(errorExtent, () => ParserStrings.InvalidUnicodeEscapeSequence);
+                return s_unknownUnicodeChar;
+            }
+
+            // Scan hex chars after the Unicode escape sequence start.
+            var sb = GetStringBuilder();
+            int i;
+            for (i = 0; i < maxNumberOfHexDigits; i++)
+            {
+                c = GetChar();
+
+                // Sequence has been terminated.
+                if (c == '}')
+                {
+                    if (i == 0)
+                    {
+                        // Sequence must have at least one hex char.
+                        IScriptExtent errorExtent = NewScriptExtent(escSeqStartIndex, _currentIndex);
+                        ReportError(errorExtent, () => ParserStrings.InvalidUnicodeEscapeSequence);
+                        return s_unknownUnicodeChar;
+                    }
+
+                    isTerminated = true;
+                    break;
+                }
+
+                if (!c.IsHexDigit())
+                {
+                    UngetChar();
+
+                    ReportError(_currentIndex, () => ParserStrings.MissingUnicodeEscapeSequenceTerminator);
+                    return s_unknownUnicodeChar;
+                }
+
+                sb.Append(c);
+            }
+
+            if (!isTerminated)
+            {
+                c = GetChar();
+                if (c != '}')
+                {
+                    UngetChar();
+                    ReportError(_currentIndex,
+                        i == maxNumberOfHexDigits && c.IsHexDigit()
+                            ? (Expression<Func<string>>)(() => ParserStrings.TooManyDigitsInUnicodeEscapeSequence)
+                            : () => ParserStrings.MissingUnicodeEscapeSequenceTerminator);
+                    return s_unknownUnicodeChar;
+                }
+            }
+
+            string hexStr = GetStringAndRelease(sb);
+
+            int unicodeValue = int.Parse(hexStr, NumberStyles.AllowHexSpecifier, NumberFormatInfo.InvariantInfo);
+            if (unicodeValue <= Char.MaxValue)
+            {
+                return ((char)unicodeValue).ToString();
+            }
+            else if (unicodeValue <= 0x10FFFF)
+            {
+                return Char.ConvertFromUtf32(unicodeValue);
```

Comment:
BTW reading through the C# impl has been enlightening.  I have been using a documented 0XFFDD as my "unknown" Unicode char but the C# compiler uses Char.MaxValue for an equivalent (I believe) "invalid" Unicode char:
```csharp
public const char InvalidCharacter = char.MaxValue;
```
Also rather than have their esc seq return a string, they return a char and then add an `out char surrogateChar` for the one specific Unicode surrogate pair case e.g.:
```csharp
private char ScanUnicodeEscape(bool peek, out char surrogateCharacter, out SyntaxDiagnosticInfo info)
```
https://github.com/dotnet/roslyn/blob/94ca919c6b4c94857dc5d58df4e6e10cb8d55c3c/src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs#L398

Which is called from this equivalent of our `Backtick`:
```csharp
private char ScanEscapeSequence(out char surrogateCharacter)
```
https://github.com/dotnet/roslyn/blob/94ca919c6b4c94857dc5d58df4e6e10cb8d55c3c/src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs#L98

And that gets called like this from the ScanStringLiteral method:
```csharp
                    if (ch == '\\' && allowEscapes)
                    {
                        // normal string & char constants can have escapes
                        char c2;
                        ch = this.ScanEscapeSequence(out c2);
                        _builder.Append(ch);
                        if (c2 != SlidingTextWindow.InvalidCharacter)
                        {
                            _builder.Append(c2);
                        }
                    }
```
https://github.com/dotnet/roslyn/blob/94ca919c6b4c94857dc5d58df4e6e10cb8d55c3c/src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs#L25

Is it worth changing the impl a bit to be more like the C# impl?  At the very least we should probably change to use the InvalidCharacter (0xffff) like C# does instead of the replacement character (0xfffd).